### ISSUE TITLE
[BUG] VoiceOver improvements

### ISSUE THREAD

**Issue by TheQuinbox** [mastodon/mastodon-ios#378]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I really appreciate your efforts to make this app more accessible to VoiceOver users, and I already see improvement. However, it's not perfect. When on the details screen, you can't read the text of the current Toot. There are a few buttons with weird names, I think the Mastodon logo? At the top of the screen. And it would be extremely nice if we could have VoiceOver actions to do a lot of common things (e.g. like, reblog, etc). In addition, when I'm composing a toot, I can't read any of the text I typed. However, keep in mind I'm on a Mac, and I know Mastodon iOS isn't officially supported there. Not sure if that one happens on iOS.

### Expected Behavior

All the aforementioned issues should be addressed, making this more the most accessible Mastodon app on the market, and from an official source, no less.

### Steps To Reproduce

Try to navigate the app using the VoiceOver screen reader.

### Environment

```markdown
- Device: MacBook Air M1 (2020)
- OS: macOS 12.3.1
- Version: 1.3.1
- Build: 113
```

**Comment by MainasuK** [comment]
Thank you for your feedback. 

There is a navigation bar title button in the home timeline. I add some label for differnet state of this button. It's will display in v1.4.0. And could you explain the how the VoiceOver action working? How can I trigger that action in Safari or others apps. I needs to learn it then try to fix it. 

In the compose and post detail page the app shoulds spells the content on iOS. I'm not try it on Mac. I'll take a notice of that.

**Comment by TheQuinbox** [comment]
To access Voice Over Actions, you use a 1 finger swipe up and down. You can try it in, for example, Twitterrific from the Icon Factory.

**Comment by ahundt** [comment]
Hi since this looks like an umbrella voiceover bug report I’m reporting the following issue encountered by a VoiceOver user. 

> I’m #Deafblind and using #VoiceOver, the iOS screenreader #blind people use. Mastodon’s button labeled Compose doesn’t work, but if I swipe right three times and double tap a button labeled “square.and chart,” then the publishing field appears! There’s a lot of visual content that’s not accessible here, too. What kind of world have I just entered?! #Accessibility #a11y

https://mastodon.social/@HabenGirma/109259674417438870

**Comment by Bracciata** [comment]
Can this be merged with #309

**Comment by j-f1** [comment]
An update on this issue:

- [x] When on the details screen, you can't read the text of the current Toot. _Fixed in #491!_
- [x] There are a few buttons with weird names, I think the Mastodon logo? At the top of the screen. _will be fixed by #784_
- [x] And it would be extremely nice if we could have VoiceOver actions to do a lot of common things (e.g. like, reblog, etc). _Added in #793!_
- In addition, when I'm composing a toot, I can't read any of the text I typed. _On macOS 12.6.1 (21G217) with the latest dev version of Mastodon, I am able to read back the text I’ve typed using the arrow keys and by selecting it using ⌘A. Please let me know / file a new issue if you’re still experiencing this_

> However, keep in mind I'm on a Mac, and I know Mastodon iOS isn't officially supported there. Not sure if that one happens on iOS.

Most of this should be fixed for macOS too but I have not done very thorough testing

**Comment by Bracciata** [comment]
@j-f1 the Mastodon Logo Button fix has been merged by the way!

**Comment by j-f1** [comment]
Thanks!
